### ISSUE TITLE
loss and accuracy are not right on kaggle-cifar and kaggle-dog

### ISSUE THREAD

**Issue by mli** [d2l-ai/d2l-en#1775]
the pytorch tab in http://d2l.ai/chapter_computer-vision/kaggle-dog.html:

![image](https://user-images.githubusercontent.com/421857/120869708-e3343280-c54b-11eb-86e3-bcfa092709df.png)

for http://d2l.ai/chapter_computer-vision/kaggle-cifar10.html

both mxnet and pytorch are not right

![image](https://user-images.githubusercontent.com/421857/120869756-00690100-c54c-11eb-8580-2b7153b3d447.png)

![image](https://user-images.githubusercontent.com/421857/120869760-04951e80-c54c-11eb-976f-01506208923c.png)

**Comment by AnirudhDagar** [comment]
@mli This is intentional. See comments on #1531 
But I think it is not very obvious at first sight and we need to explicitly mention this. What do you think @astonzhang ?

**Comment by mli** [comment]
i'll suggest to have reasonable HP, otherwise audience may think there are bugs in the code. We should avoid  not very obvious at first sight things, as it causes confusions and eats our bandwidth to explain.

**Comment by astonzhang** [comment]
The demo datasets are both very small in these cases to save execution time. At the very least, we'll need to show declining training loss over time. The PyTorch plot misses the training loss due to random NAN error. Let me look into it.

**Comment by mli** [comment]
I feel most users will tune based on the small datasets, instead of the
full dataset. so having a good default HP for the small one matters

**Comment by astonzhang** [comment]
Totally agree - the original Kaggle data are very big and expensive to play with. 

I've revised implementations in these two sections. Now on the local small-scale demo datasets we still have decent plots:

<img width="1123" alt="Screen Shot 2021-06-04 at 10 06 37 PM" src="https://user-images.githubusercontent.com/22279212/120880835-89e6f600-c581-11eb-9c69-5ab708859039.png">

<img width="1101" alt="Screen Shot 2021-06-04 at 10 06 45 PM" src="https://user-images.githubusercontent.com/22279212/120880836-8ce1e680-c581-11eb-9540-9e67fd06729d.png">

<img width="1098" alt="Screen Shot 2021-06-04 at 10 18 49 PM" src="https://user-images.githubusercontent.com/22279212/120881011-dda60f00-c582-11eb-996a-172214d5ccf8.png">

<img width="1107" alt="Screen Shot 2021-06-04 at 10 18 57 PM" src="https://user-images.githubusercontent.com/22279212/120881015-e4cd1d00-c582-11eb-9dd3-1f6ee7d9d13f.png">


@AnirudhDagar Here are my relevant updates:
https://github.com/d2l-ai/d2l-en/commit/aa2a9a3e041b684ae434a9af470e8aa92aa26fb8
https://github.com/d2l-ai/d2l-en/commit/2827fe50b227052bbc5b5ebc8948baebac955da1
https://github.com/d2l-ai/d2l-en/commit/8b325e369d3723fd3d700a2c5356afcb135ff818

which haven't touched the PyTorch part. Could you adapt my above changes to PyTorch? Thanks.

@goldmermaid - After Anirudh made such changes, could you quickly adapt them into the Chinese version. Then I'll re-build both release branches before Mu's mooc on these Kaggle competitions. Thanks!

**Comment by astonzhang** [comment]
A downside for such changes is that our exercises become slightly more verbose to describe a decent set of HPs for the full dataset. Since we expect that most users play with the demo small datasets, this should be ok.

**Comment by goldmermaid** [comment]
> @AnirudhDagar Here are my relevant updates:
> [aa2a9a3](https://github.com/d2l-ai/d2l-en/commit/aa2a9a3e041b684ae434a9af470e8aa92aa26fb8)
> [2827fe5](https://github.com/d2l-ai/d2l-en/commit/2827fe50b227052bbc5b5ebc8948baebac955da1)
> [8b325e3](https://github.com/d2l-ai/d2l-en/commit/8b325e369d3723fd3d700a2c5356afcb135ff818)
> 
> which haven't touched the PyTorch part. Could you adapt my above changes to PyTorch? Thanks.
> 
> @goldmermaid - After Anirudh made such changes, could you quickly adapt them into the Chinese version. Then I'll re-build both release branches before Mu's mooc on these Kaggle competitions. Thanks!

d2l-zh fixes [here](https://github.com/d2l-ai/d2l-zh/pull/855).

@AnirudhDagar please let me know once you are done with HP tuning. Thanks!

**Comment by AnirudhDagar** [comment]
Hi! Sorry for the late reply on this. Have been busy throughout this week. I'll send PRs to fix both these sections.

**Comment by AnirudhDagar** [comment]
@goldmermaid Please go ahead with the fixes in d2l-zh. Again sorry for the delay caused.

**Comment by astonzhang** [comment]
Thanks! I also made very minor revision: https://github.com/d2l-ai/d2l-en/commit/0da9c0646ddc6e6c8c9512296414ccbf4bdb9857

@goldmermaid Now these two sections are ready to be ported to d2l-zh

**Comment by goldmermaid** [comment]
> Thanks! I also made very minor revision: [0da9c06](https://github.com/d2l-ai/d2l-en/commit/0da9c0646ddc6e6c8c9512296414ccbf4bdb9857)
> 
> @goldmermaid Now these two sections are ready to be ported to d2l-zh

Migration done at https://github.com/d2l-ai/d2l-zh/pull/855!
